code/datums/components/stationstuck.dm

```diff
@@ -0,0 +1,41 @@
+//very similar to stationloving, but more made for mobs and not objects. used on derelict drones currently
+/datum/component/stationstuck
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/murder = TRUE //teleports if not
+	var/stuck_zlevel
+	var/teleport_message = ""
+	var/death_message = ""
+
+/datum/component/stationstuck/Initialize(_murder = TRUE, _teleport_message = "", _death_message = "")
```

Comment:
You just need a single message argument since only one will ever be used depending on `_murder`

Reply:
i like this idea